atlante/server/server.go

```diff
@@ -196,6 +233,106 @@ func encodeCellAsJSON(w io.Writer, cell *grids.Cell, pdf string, lat, lng *float
 	json.NewEncoder(w).Encode(jsonCell)
 }
 
+// CreateJobDB will create the job tracking database
+func CreateJobDB(filename string) (*sql.DB, error) {
+	database, err := sql.Open("sqlite3", filename)
+	if err != nil {
+		return nil, err
+	}
+	statement, err := database.Prepare(`
+	CREATE TABLE IF NOT EXISTS jobs (
+		id INTEGER PRIMARY KEY, 
+		job_id TEXT, 
+		mdgid TEXT,
+		status TEXT,
+		enqued_at TEXT,
+		updated_at TEXT,
+		qjob_id TEXT,
+		jobobj TEXT
+	)
+`)
+	if err != nil {
+		return nil, err
+	}
+	statement.Exec()
+	statement, err = database.Prepare(`CREATE UNIQUE INDEX idx_job_id ON jobs (job_id);`)
+	if err != nil {
+		return nil, err
+	}
+	statement.Exec()
+	statement, err = database.Prepare(`CREATE UNIQUE INDEX idx_qjob_id ON jobs (qjob_id);`)
+	if err != nil {
+		return nil, err
+	}
+	statement.Exec()
+	return database, nil
+}
+
+func (s *Server) findJobItem(mdgidstr string) (jbs []jobItem, err error) {
+	database := s.jobsDB
+	if database == nil {
+		return nil, nil
+	}
+	mdgid := grids.NewMDGID(mdgidstr)
+	const selectSQL = `
+	SELECT  
+		id,
+		job_id,
+		status,
+		enqued_at,
+		updated_at
+	FROM 
+		jobs
+	WHERE 
+		mdgid=?
+	ORDER BY id DESC
+	`
+	rows, err := database.Query(selectSQL, mdgid.AsString())
+	if err != nil {
+		return jbs, err
+	}
+
+	for rows.Next() {
+		var enquedat, updatedat string
+		ji := jobItem{MdgID: mdgid.Id}
+		if mdgid.Part != 0 {
+			ji.MdgIDPart = mdgid.Part
+		}
+		err = rows.Scan(
+			&ji.ID,
+			&ji.JobID,
+			&ji.Status,
+			&enquedat,
+			&updatedat,
+		)
+		if err != nil {
+			return []jobItem{}, err
+		}
+		ji.EnquedAt, err = time.Parse(time.RFC3339, enquedat)
+		if err != nil {
+			log.Warnf("failed to parse enqued at from db: %v for id: %v", enquedat, ji.ID)
+		}
+		ji.UpdatedAt, err = time.Parse(time.RFC3339, updatedat)
+		if err != nil {
+			log.Warnf("failed to parse updated at from db: %v for id: %v", updatedat, ji.ID)
+		}
+		jbs = append(jbs, ji)
+	}
+	return jbs, nil
+}
+
+// TODO(gdey): job management
+func (s *Server) addJobItem(ji *jobItem) (*jobItem, error) {
+	database := s.jobsDB
+	if database == nil {
+		return nil, fmt.Errorf("jobDB not initilized")
+	}
+	const intertSQL = `
+	INSERT INTO jobs (job_id, status, ) VALUES (?, ?)
```

Comment:
This SQL statement looks broken with the extra comma after `status`. Looks like this might still be in the works.